src/tracks/playabletrack/wavetrack/ui/SpectrumView.cpp

```diff
@@ -333,7 +332,8 @@ void DrawClipSpectrum(TrackPanelDrawingContext &context,
       return;
    }
 
-   const WaveTrack *const track = waveTrackCache.GetTrack().get();
+   const auto track =
+      static_cast<const WaveTrack*>(waveTrackCache.GetTrack().get());
```

Comment:
Is it really safe?

Reply:
I thought you would flag that.

If we assume no other concrete subclasses of SampleTrack exist, yes.

Will that be a safe future assumption? ... Maybe not.

Reply:
Would it be nice to make SpectrumView depend only on SampleTrack, not WaveTrack?  Yes.

What stops it here?  The need for the track to define GetSpectrogramSettings and GetSpectrumBounds.

I have other plans, later in the big branch, to move those things into ClientData attachments to the SampleTrack, but that is a bit beside the point of the present library extraction.

Reply:
I could make it a dynamic_cast, and return early if it is null.  That means if we attempt spectrogram display of some other kind of hypothetical future concrete SampleTrack, it would appear as a blank rectangle, but not be a crashing bug.

Reply:
Do you not object to the other static_cast in WaveClip.cpp?  (Those things only ever belong to WaveTracks.)

Have you reviewed all else and found no problems?

Reply:
SampleTrack does not expose the notion of division of the sample sequence into clips.  It might need that for generalizing SpectrumView in future.  It is not yet needed for purposes of extracting AudioIO so it does not depend on WaveTrack.

Reply:
I force-pushed a fix for your complaint.